src/chat.ts

```diff
@@ -36,24 +68,40 @@ export class ChatProvider implements vscode.WebviewViewProvider {
       }
     });
 
-    webviewView.webview.onDidReceiveMessage(async (data) => {
+    webviewView.webview.onDidReceiveMessage(async (data: ChatRequest) => {
       switch (data.type) {
         case "chat_request": {
+          this.currentAssistantMessage = "";
           vscode.commands.executeCommand("sourcery.chat_request", data);
           break;
         }
       }
     });
   }
 
-  public addResult(result) {
+  public addResult(result: ChatResult) {
+    // Send the whole message we've been streamed so far to the webview,
+    // after converting from markdown to html
+
+    if (result.outcome === ChatResultOutcome.Success) {
+      this.currentAssistantMessage += result.textContent;
+    }
```

Comment:
build up the whole current message here so we can convert the whole thing to html and send it each time we get a streaming response